### PR TITLE
checking if firewalld service is running

### PR DESCRIPTION
**Pull request by devidask27**
Bailing out if firewalld service is running on centos
Screenshot 2021-08-05 at 3 46 26 PM

### CODE REVIEW

File: pkg/platform/centos/centos.go [hunk 1]
```diff
@@ -334,3 +337,12 @@ func (c *CentOS) checkPIDofSystemd() (bool, error) {
 		return true, nil
 	}
 }
+
+func (c *CentOS) checkFirewalldIsRunning() (bool, error) {
+	_, err := c.exec.RunWithStdout("bash", "-c", "systemctl status firewalld | grep 'active (running)'")
```
Comment (sahil-lakhwani) [inline pkg/platform/centos/centos.go:342]: do we need `bash -c` ?

Comment (devidask27) [reply pkg/platform/centos/centos.go:342]: Yes

File: pkg/platform/centos/centos.go [hunk 2]
```diff
@@ -69,6 +69,9 @@ func (c *CentOS) Check() []platform.Check {
 	result, err = c.checkPIDofSystemd()
 	checks = append(checks, platform.Check{"Check if system is booted with systemd", true, result, err, fmt.Sprintf("%s", err)})
 
+	result, err = c.checkFirewalldIsRunning()
+	checks = append(checks, platform.Check{"Check if firewalld service is running", true, result, err, fmt.Sprintf("%s", err)})
```
Comment (anupbarve) [inline pkg/platform/centos/centos.go:73]: This check needs to be optional.

Comment (devidask27) [reply pkg/platform/centos/centos.go:73]: Okay

File: pkg/platform/centos/centos.go (hunk 1 above)
Comment (anupbarve) [inline pkg/platform/centos/centos.go:342]: You could check if the following command is useful
`systemctl is-active firewalld`

**Review by anupbarve — COMMENTED** [review]
I was able to install firewalld on my ubuntu machine as well.
We need to cover debian based checks as well.

**Pull request merged**